db/compaction_picker.cc

```diff
@@ -1373,6 +1356,39 @@ bool LevelCompactionBuilder::PickFileToCompact() {
   return start_level_inputs_.size() > 0;
 }
 
+bool LevelCompactionBuilder::PickL0ToBaseCompaction() {
```

Comment:
Or we can consider checking it in as-is. People can't rely on the mentioned optimization anyways since it requires the largest L0 file to minimally overlap with L1, which they can't control. This L0 compaction picking is a lot simpler, especially for use together with intra-L0 compaction, in my view.